src/mjolnir/graphenhancer.cc

```diff
@@ -614,15 +637,18 @@ bool IsNotThruEdge(GraphReader& reader,
   const GraphTile* tile;
   for (uint32_t n = 0; n < kMaxNoThruTries; n++) {
     // If expand list is exhausted this is "not thru"
-    if (expandset.empty()) {
+    if (expand_pos == expandset.size()) {
       return true;
     }
 
     // Get the node off of the expand list and add it to the visited list.
-    // Expand edges from this node.
-    const GraphId expandnode = *expandset.cbegin();
-    expandset.erase(expandset.begin());
+    // Expand edges from this node.  Post-increment the index to the next
+    // item.
+    const GraphId expandnode = expandset[expand_pos++];
```

Comment:
Wonder if you can use `GraphId.value` as a key to your bloom filter? (will they be unique in this context?)

Reply:
@mandeepsandhu I had that initially - but I think because many edges will be within the same tile, and not many bits change between values, we would end up with bloom filter collisions really quickly.  I added hashing to shake things around a bit.

Reply:
Or do you mean still hash, but only use the `.value()` part of the GraphId?

That might collide if we're exploring a location that crosses tile boundaries.  We definitely don't need to be perfectly collision avoiding here, just better-than-nothing.

Reply:
No I meant using `GraphId.value` directly without running through the hash function. But yeah, if there are a lot of collisions it will cause extra hash lookups which might not be as effective.